### ISSUE TITLE
Wrong diagnostic: `return type notation`

### ISSUE THREAD

**Issue by dev-ardi** [rust-lang/rust#134087]
### Code

```Rust
`let x: Vec::new();`
```

### Current output

```Shell
error: return type notation arguments must be elided with `..`
 --> src/main.rs:2:20
  |
2 |     let x: Vec::new();
  |                    ^^ help: add `..`: `(..)`

error: return type notation not allowed in this position yet
 --> src/main.rs:2:12
  |
2 |     let x: Vec::new();
  |            ^^^^^^^^^^
```

### Desired output

```Shell

```

### Rationale and extra context

This is a mistake where `:` was used instead of `=`. Instead of telling the user that it talks about rtn.

### Other cases

```Rust

```

### Rust Version

```Shell
stable and latest nightly.
rustc 1.85.0-nightly (a2545fd6f 2024-11-28)
```

### Anything else?

@rustbot D-invalid-suggestion

**Comment by RodBurman** [comment]
I got uneasy when I read the rationale here since given:

let x: Vec::new();

The rationale says the writer could only have "really" meant:

let x = Vec::new();  // inferred x's type

Which  I agree is most likely but people (and IDE's) are not that simple so another possibility is:

let x: appropriateType = Vec::new();

(Maybe not likely with Vec but in the wider case of x = someType::someFunction() maybe more so)

So I think we need to be careful how we improve the diagnostic lest after the improvement's release another issue is raised "Rust said just change : to =" but that is a very poor diagnostic with <new code example here>.
